src/main/kotlin/pt/up/fe/ni/website/backend/service/ErrorMessages.kt

```diff
@@ -11,6 +11,12 @@ object ErrorMessages {
 
     const val expiredRefreshToken = "refresh token has expired"
 
+    const val invalidRecoveryToken = "invalid password recovery token"
+
+    const val expiredRecoveryToken = "password recovery token has expired"
+
+    const val missingHashClaim = "password hash claim is missing"
```

Comment:
I'm thinking if we should return a general `invalid recovery token` error instead of specifying it. This may be a detail we want to omit from users, wdyt @DoStini ?

Reply:
I did it that way because it would only error if it's not implemented well in the code (forgetting to set `usePasswordHash` to true and use it). But if it eventually occurs with the user, it makes sense to omit it.

Reply:
I was just reading this now, I've mentioned in my previous comments that we should omit these kinds of errors to the users so that information on who uses the website doesn't get leaked. Always return ok even if the account doesn't exist

Reply:
Now that I'm reading this again, it makes sense to have an `invalid recovery token` error whenever we try to set a new password.